enhancements/sig-policy/98-long-term-compliance-history/README.md

```diff
@@ -0,0 +1,455 @@
+
+There would be a combined unique constraint on the `kind`, `api_group`, `name`, `parent_policy_name`, and `spec_hash`
+columns. `spec_hash` would have an index as well.
+
+The `spec` of the policy (e.g. `ConfigurationPolicy`) is stored for a customer to audit the policy contents that
+generated a compliance event. This would be stored as JSON with no new lines and no optional spaces for size efficiency.
+
+The `spec_hash` field would be used for efficient querying to find the `policy_id` value for a `compliance_events` table
+entry. It is also an efficient way to determine uniqueness of a spec for data deduplication in this table. The SHA1 hash
+would be calculated from the `spec` column value.
+
+##### compliance_events Table
+
+| Column     | Description                                                                               |
+| ---------- | ----------------------------------------------------------------------------------------- |
+| id         | an internal primary key                                                                   |
+| cluster_id | a foreign key to the `id` column in the `clusters` table                                  |
+| policy_id  | a foreign key to the `id` column in the `policies` table                                  |
+| compliance | a string column of a compliance state name (i.e. Compliant); this will have an index      |
+| message    | a string column with the compliance message                                               |
+| timestamp  | a UTC date time column for the timestamp of the compliance event; this will have an index |
+| metadata   | a JSON (or string in SQLite) column with additional metadata                              |
+
+The `metadata` field would be empty for now but could be eventually used to store information such as the object diff of
+an inform `ConfigurationPolicy` without altering the database schema. This field is not designed for filtering when
+generating reports.
+
+#### How a Compliance Event is Recorded
+
+##### Current State
+
+A policy controller, such as the Configuration Policy controller, will evaluate a policy (e.g. a `ConfigurationPolicy`)
+and produce a compliance event. This is currently recorded as a Kubernetes event. The Status Sync controller will
+process the Kubernetes event and add the status entry in the local and hub replicated policy that owns the evaluated
+policy (i.e. in the `policy-templates` array). A policy's history is limited to 10 events.
+
+##### Recording a Compliance Event
+
+As previously mentioned, the compliance events will be stored in a relational database. This database will not be
+managed by Open Cluster Management and is limited to just Postgresql at the momement. The connection information for
+this database will be in a secret called `governance-policy-database` in the `open-cluster-management` namespace and
+require a key of `connectionURL` in the format of `postgres://username:password@localhost:5432/database_name`. For
```

Comment:
Customers may want to use ESO so having a secret with username and password directly in the secret instead of that URL format I think would be needed.